src/hooks/use-snapshot.ts

```diff
@@ -12,38 +12,27 @@ interface UseSnapshotProps {
 }
 
 const useSnapshot = ({ layoutService, viewService, rect, model }: UseSnapshotProps): stardust.Rect => {
-  onTakeSnapshot(async (copyOfLayout: SnapshotLayout) => {
-    if (!copyOfLayout.snapshotData) {
-      return copyOfLayout;
+  onTakeSnapshot(async (snapshotLayout: SnapshotLayout) => {
+    if (!snapshotLayout.snapshotData || !model || snapshotLayout.snapshotData.content) {
+      return snapshotLayout;
     }
 
-    if (!model) {
-      return copyOfLayout;
-    }
-
-    if (!copyOfLayout.snapshotData.content) {
-      if ((model as EngineAPI.IGenericObject)?.getHyperCubePivotData) {
-        const pivotPages = await (model as EngineAPI.IGenericObject).getHyperCubePivotData(Q_PATH, [
+    if ((model as EngineAPI.IGenericObject)?.getHyperCubePivotData) {
+      snapshotLayout.snapshotData.content = {
+        qPivotDataPages: await (model as EngineAPI.IGenericObject).getHyperCubePivotData(Q_PATH, [
           {
             qLeft: viewService.gridColumnStartIndex,
             qTop: viewService.gridRowStartIndex,
             qWidth: viewService.gridWidth,
             qHeight: viewService.gridHeight,
           },
-        ]);
-
-        copyOfLayout.snapshotData.content = {
-          qPivotDataPages: pivotPages,
-        };
-      }
+        ]),
+      };
 
-      copyOfLayout.snapshotData.object.size.w = rect.width;
-      copyOfLayout.snapshotData.object.size.h = rect.height;
-
-      return copyOfLayout;
+      snapshotLayout.snapshotData.object.size.w = rect.width;
+      snapshotLayout.snapshotData.object.size.h = rect.height;
     }
-
-    return copyOfLayout;
+    return snapshotLayout;
   });
 
   if (layoutService.layout.snapshotData?.content) {
```

Comment:
I'm not sure when this condition is used ?! 🤔

Reply:
Don't quite remember but could be when the chart is re-sized in storytelling

Reply:
Did some testing and cannot find a case where this logic is triggered. For now lets keep it but we might be able to remove it in the feature